corim/example_profile_test.go

```diff
@@ -0,0 +1,199 @@
+	}
+
+	if *c.Language != "en-GB" {
+		return fmt.Errorf(`language must be "en-GB", but found %q`, *c.Language)
+	}
+
+	return nil
+}
+
+// Registering the profile inside init() in the same file where it is defined
+// ensures that the profile will always be available, and you don't need to
+// remember to register it at the time you want to use it. The only potential
+// danger with that is if the your profile ID clashes with another profile,
+// which should not happen if it a registered PEN or a URL containing a domain
+// that you own.
+func init() {
+	profileID, err := eat.NewProfile("http://example.com/example-profile")
+	if err != nil {
+		panic(err) // will not error, as the hard-coded string above is valid
+	}
+
+	extMap := extensions.NewMap().
+		Add(ExtEntity, &EntityExtensions{}).
+		Add(comid.ExtComid, &ComidExtensions{}).
+		Add(comid.ExtEntity, &EntityExtensions{}).
+		Add(comid.ExtReferenceValue, &RefValExtensions{})
+
+	if err := RegisterProfile(profileID, extMap); err != nil {
+		// will not error, assuming our profile ID is unique, and we've
+		// correctly set up the extensions Map above
+		panic(err)
+	}
+}
+
+// ----- end of profile definition -----
+// The following code demonstrates how the profile might be used.
+
+func Example_profile_unmarshal() {
+	buf, err := os.ReadFile("testcases/unsigned-example-corim.cbor")
+	if err != nil {
+		log.Fatalf("could not read corim file: %v", err)
+	}
+
+	// UnmarshalUnsignedCorimFromCBOR will detect the profile and ensure
+	// the correct extensions are loaded before unmarshalling
+	extractedCorim, err := UnmarshalUnsignedCorimFromCBOR(buf)
+	if err != nil {
+		log.Fatalf("could not unmarshal corim: %v", err)
+	}
+
+	extractedComid, err := UnmarshalComidFromCBOR(
+		extractedCorim.Tags[0],
+		extractedCorim.Profile,
+	)
+	if err != nil {
+		log.Fatalf("could not unmarshal corim: %v", err)
+	}
+
+	fmt.Printf("Language: %s\n", *extractedComid.Language)
+	fmt.Printf("Entity: %s\n", *extractedComid.Entities.Values[0].Name)
+	fmt.Printf("        %s\n", extractedComid.Entities.Values[0].
+		Extensions.MustGetString("Address"))
+
+	fmt.Printf("Measurements:\n")
+	for _, m := range extractedComid.Triples.ReferenceValues.Values[0].Measurements.Values {
+
+		val := hex.EncodeToString((*m.Val.Digests)[0].HashValue)
+		tsInt := m.Val.Extensions.MustGetInt64("timestamp")
+		ts := time.Unix(tsInt, 0).UTC()
+
+		fmt.Printf("    %v taken at %s\n", val, ts.Format("2006-01-02T15:04:05"))
+	}
+
+	// output:
+	// Language: en-GB
+	// Entity: ACME Ltd.
+	//         123 Fake Street
+	// Measurements:
+	//     87428fc522803d31065e7bce3cf03fe475096631e5e07bbd7a0fde60c4cf25c7 taken at 2024-07-12T11:03:10
+	//     0263829989b6fd954f72baaf2fc64bc2e2f01d692d4de72986ea808f6e99813f taken at 2024-07-12T11:03:10
+	//     a3a5e715f0cc574a73c3f9bebb6bc24f32ffd5b67b387244c2c909da779a1478 taken at 2024-07-12T11:03:10
+}
+
+// note: this example is rather verbose as we're going to be constructing a
```

Comment:
```suggestion
// note: this example is rather verbose as we're going to construct a
```

Reply:
What's wrong with the existing phrasing? The imperfect tense seems more appropriate, as the comment is concerned with the process, not the end result (i.e. the point isn't that we're gonna construct a CoMID, but that we will be doing so by hand).

Reply:
That is fine, then you want to say:
this example is rather verabose as we are constructing a

Reply:
As the comment before the example, we haven't started yet, so we're going to be doing that, not presently doing that.

Your suggested phrasing would make sesne if the comment was in the middle of the example.